pkg/lib/aws/servicequotas.go

```diff
@@ -159,181 +152,126 @@ func (c *Client) VerifyInstanceQuota(instances []InstanceTypeRequests) error {
 	return nil
 }
 
-func (c *Client) VerifyNetworkQuotas(
-	requiredInternetGateways int,
-	natGatewayRequired bool,
-	highlyAvailableNATGateway bool,
-	requiredVPCs int,
-	availabilityZones strset.Set,
-	numNodeGroups int,
-	longestCIDRWhiteList int) error {
-	quotaCodeToValueMap := map[string]int{
-		_elasticIPsQuotaCode:         0, // elastic IP quota code
-		_internetGatewayQuotaCode:    0, // internet gw quota code
-		_natGatewayQuotaCode:         0, // nat gw quota code
-		_vpcQuotaCode:                0, // vpc quota code
-		_securityGroupsQuotaCode:     0, // security groups quota code
-		_securityGroupRulesQuotaCode: 0, // security group rules quota code
-	}
-
-	err := c.ServiceQuotas().ListServiceQuotasPages(
-		&servicequotas.ListServiceQuotasInput{
-			ServiceCode: aws.String("ec2"),
-		},
-		func(page *servicequotas.ListServiceQuotasOutput, lastPage bool) bool {
-			if page == nil {
-				return false
-			}
-			for _, quota := range page.Quotas {
-				if quota == nil || quota.QuotaCode == nil || quota.Value == nil {
-					continue
-				}
-				if _, ok := quotaCodeToValueMap[*quota.QuotaCode]; ok {
-					quotaCodeToValueMap[*quota.QuotaCode] = int(*quota.Value)
-					return false
-				}
-			}
-			return true
-		},
-	)
+func (c *Client) VerifyInternetGatewayQuota(internetGatewayQuota int, requiredInternetGateways int) error {
+	internetGatewaysInUse, err := c.ListInternetGateways()
 	if err != nil {
-		return errors.WithStack(err)
+		return err
+	}
+	if internetGatewayQuota-len(internetGatewaysInUse)-requiredInternetGateways < 0 {
+		additionalQuotaRequired := len(internetGatewaysInUse) + requiredInternetGateways - internetGatewayQuota
```

Comment:
Since `additionalQuotaRequired` is the negative of `internetGatewayQuota-len(internetGatewaysInUse)-requiredInternetGateways`, we can just move `additionalQuotaRequired` outside of the if branch and do `if additionalQuotaRequired > 0`.

Same for `VerifyEIPQuota`, `VerifyVPCQuota` and `VerifySecurityGroupQuota` functions (where applicable).